include/seqan/graph_msa/graph_align_tcoffee_msa.h

```diff
@@ -471,17 +456,33 @@ void globalMsaAlignment(Graph<Alignment<TStringSet, TCargo, TSpec> > & gAlign,
         }
     }
 
+    // Re-score the matches.
+    TGraph g(seqSet);
+    if (msaOpt.rescore) // in current implementation always true
+        _scoreMatches(stringSet(g), msaOpt.sc, matches, scores);
+
+    // Include a T-Coffee library.
+    if (!empty(msaOpt.libfiles))
+    {
+        typedef typename Iterator<String<std::string> const, Standard>::Type TIter;
+        TIter begIt = begin(msaOpt.libfiles, Standard());
+        TIter begItEnd = end(msaOpt.libfiles, Standard());
+        for (; begIt != begItEnd; goNext(begIt))
+        {
+            std::ifstream strm_lib;
+            strm_lib.open((*begIt).c_str(), std::ios_base::in | std::ios_base::binary);
+            read(strm_lib, matches, scores, sequenceNames, TCoffeeLib());
+            strm_lib.close();
+        }
+    }
+
 #ifdef SEQAN_TCOFFEE_DEBUG
     std::cout << std::setw(30) << std::left << "Segment-match generation:" << std::setw(10) << std::right << sysTime() - segmentGenerationTime << "  s" << std::endl;
     std::cout << std::setw(30) << std::left << "Number of segment-matches:" << std::setw(10) << std::right << length(matches) << std::endl;
 #endif
 
-    // Use these segment matches for the initial alignment graph
-    TGraph g(seqSet);
-    if (!msaOpt.rescore)
-        buildAlignmentGraph(matches, scores, g, FractionalScore());
-    else
-        buildAlignmentGraph(matches, scores, g, msaOpt.sc, ReScore());
+    // Use these segment matches for the initial alignment graph.
+    buildAlignmentGraph(matches, scores, g, FractionalScore());
```

Comment:
quick questions:
If `msaOpt.rescore` is always true, we never called `buildAlignmentGraph(matches, scores, g, FractionalScore());` but now we always do. Is that always correct? Does the rescoring has any effect if  `buildAlignmentGraph(matches, scores, g, FractionalScore());` is called afterwards anyway?

Reply:
Good questions! 

(1) Yes, that is always correct. The function `buildAlignmentGraph(..., ReScore())` does exactly two things: `scoreMatches` and then `buildAlignmentGraph(..., FractionalScore())` (see its implementation in graph_align_tcoffee_base.h, L.262). The re-scoring I evaluate before the lib-files are read, the actual graph building is processed afterwards. 

(2) Yes, because it modifies the score values that are later used as edge weights in the graph. It ensures that e.g. there are no negative values.